pkg/deploy/template_chart_manager.go

```diff
@@ -0,0 +1,146 @@
+	if err != nil {
+		return true, err
+	}
+	manifest, err := renderChartToRawManifest(ctx, kymaObjKey, m.config.WebhookChartPath, chartArgValues)
+	if err != nil {
+		return true, err
+	}
+	resources, err := getRawManifestUnstructuredResources(manifest)
+	if err != nil {
+		return true, err
+	}
+	for _, resource := range resources {
+		if err := createOrUpdateResource(ctx, syncContext.RuntimeClient, resource); err != nil {
+			return true, err
+		}
+	}
+	kyma.UpdateCondition(v1alpha1.ConditionReasonSKRWebhookIsReady, metav1.ConditionTrue)
+	logger.Info("successfully installed webhook chart",
+		"release-name", skrChartReleaseName(kymaObjKey))
+	return false, nil
+}
+
+func (m *SKRWebhookTemplateChartManager) Remove(ctx context.Context, kyma *v1alpha1.Kyma) error {
+	logger := logf.FromContext(ctx)
+	kymaObjKey := client.ObjectKeyFromObject(kyma)
+	syncContext := remote.SyncContextFromContext(ctx)
+	manifest, err := renderChartToRawManifest(ctx, kymaObjKey, m.config.WebhookChartPath, map[string]interface{}{})
+	if err != nil {
+		return err
+	}
+	resources, err := getRawManifestUnstructuredResources(manifest)
+	if err != nil {
+		return err
+	}
+	for _, resource := range resources {
+		if err := syncContext.RuntimeClient.Delete(ctx, resource); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete webhook %s: %w", resource.GetKind(), err)
+		}
+	}
+	logger.Info("successfully removed webhook chart",
+		"release-name", skrChartReleaseName(kymaObjKey))
+	return nil
+}
+
+func getRawManifestUnstructuredResources(rawManifest string) ([]*unstructured.Unstructured, error) {
+	stringReader := strings.NewReader(rawManifest)
+	k8syaml.NewYAMLOrJSONDecoder(stringReader, defaultBufferSize)
+	decoder := k8syaml.NewYAMLOrJSONDecoder(stringReader, defaultBufferSize)
+	var resources []*unstructured.Unstructured
+	for {
+		resource := &unstructured.Unstructured{}
+		err := decoder.Decode(resource)
+		if err != nil && !errors.Is(err, io.EOF) {
+			return nil, err
+		}
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		resources = append(resources, resource)
+	}
+	return resources, nil
+}
+
+func createOrUpdateResource(ctx context.Context, skrClient client.Client,
+	resource *unstructured.Unstructured,
+) error {
+	err := skrClient.Create(ctx, resource)
+	resourceAlreadyExists := apierrors.IsAlreadyExists(err)
+	if err != nil && !resourceAlreadyExists {
+		return fmt.Errorf("failed to create webhook %s: %w", resource.GetKind(), err)
+	}
+	if resourceAlreadyExists {
+		resourceObjKey := client.ObjectKeyFromObject(resource)
+		oldResource := &unstructured.Unstructured{}
+		oldResource.SetGroupVersionKind(resource.GroupVersionKind())
+		if err := skrClient.Get(ctx, resourceObjKey, oldResource); err != nil {
+			return fmt.Errorf("failed to get webhook %s: %w", resource.GetKind(), err)
+		}
+		resource.SetResourceVersion(oldResource.GetResourceVersion())
```

Comment:
why setting this explicitly?

Reply:
In CSA, without the resource version field set, the `Update()` call fails. As agreed already in the daily, this implementation will be replaced with the SSA implementation.